lib/gen_socket_client.ex

```diff
@@ -155,7 +154,7 @@ defmodule Phoenix.Channels.GenSocketClient do
         {:error, :already_joined}
       true ->
         frame = transport.serializer.encode_message(%{topic: topic, event: event, payload: payload, ref: ref})
-        WebSocketClient.push(transport.transport_pid, frame)
+        transport.transport.push(transport.transport_pid, frame)
```

Comment:
I wonder if `transport` is a good name here. `transport.transport` certainly is very clunky.
The first `transport` is more of a meta data structure of sorts, right? Maybe it should be named accordingly? Or maybe it's just really a `state` structure of sorts?